tools/python/data-generator/flight_data_generator.py

```diff
@@ -0,0 +1,408 @@
+                            "Value": str(v),
+                            "DimensionValueType": "VARCHAR"
+                        })
+
+                measures = []
+                for measure_template in self.measure_templates:
+                    measure_name = measure_template["name"]
+                    measure_value_type = measure_template.get("type", "DOUBLE").upper()
+
+                    if measure_name in ["lat", "lon"]:
+                        measure_value = self._update_entity_position(entity, measure_name, reporting_frequency)
+                    elif measure_name == "heading":
+                        # Calculate heading based on current and target positions
+                        measure_value = self._calculate_bearing(entity["current_lat"], entity["current_lon"],
+                                                                entity["target_lat"], entity["target_lon"])
+                    else:
+                        measure_value = self._generate_measure_value(
+                            measure_template, entity, current_date == start_date
+                        )
+
+                    # Save for next iteration
+                    entity["latest_measures"][measure_name] = measure_value
+
+                    measures.append({
+                        "MeasureName": measure_name,
+                        "MeasureValueType": measure_value_type,
+                        "MeasureValue": str(measure_value)
+                    })
+
+                precision = precision.strip().upper()
+                if precision == "SECONDS":
+                    timestamp = str(int(current_date.timestamp()))
+                if precision == "MICROSECONDS":
+                    timestamp = str(int(current_date.timestamp() * 1_000_000))
+                else:
+                    # Default to millisecond precision
+                    timestamp = str(int(current_date.timestamp() * 1_000))
+
+                record = {
+                    "Dimensions": dimensions,
+                    "Time": timestamp,
+                    "Measures": measures
+                }
+                records.append(record)
+
+            current_date += reporting_frequency
+        return records
+
+    def _interpolate_great_circle(self, lat1, lon1, lat2, lon2, fraction):
+        """
+        Interpolates between two points (lat1, lon1) and (lat2, lon2) along the great-circle path.
+
+        :param lat1: Starting latitude in degrees
+        :param lon1: Starting longitude in degrees
+        :param lat2: Target latitude in degrees
+        :param lon2: Target longitude in degrees
+        :param fraction: Fraction of the distance to interpolate (0.0 to 1.0)
+        :return: (latitude, longitude) tuple of the interpolated point
+        """
+        phi1 = math.radians(lat1)
+        phi2 = math.radians(lat2)
+        lambda1 = math.radians(lon1)
+        lambda2 = math.radians(lon2)
+
+        # Compute the angular distance between the points
```

Comment:
```suggestion
        # Compute the angular distance between the points using the spherical law of cosines
```

More of an explanation would make understanding the formulas easier.

Reply:
Updated the code to use the [great_circle_calculator](https://github.com/seangrogan/great_circle_calculator) package, which uses the formulas from [here](https://www.movable-type.co.uk/scripts/latlong.html) and [here](https://www.edwilliams.org/avform147.htm) according to their docs.

Reply:
Awesome, good idea.